### PR TITLE
Adapt lights simultaneously instead of one by one

### PR DESCRIPTION
**Pull request by th3w1zard1**
For users using high values of `send_split_delay`, it's very noticeable how multiple lights in a switch turn on one after another.

### CODE REVIEW

**Comment by th3w1zard1** [comment]
I don't know how such a simple change doesn't pass the test.

File: tests/test_switch.py [hunk 1]
```diff
@@ -595,7 +595,7 @@ def increased_color_temp():
 
 async def test_auto_reset_manual_control(hass):
     switch, (light, *_) = await setup_lights_and_switch(
-        hass, {CONF_AUTORESET_CONTROL: 0.1}
+        hass, {CONF_AUTORESET_CONTROL: 0.2}
```
Comment (basnijholt) [inline tests/test_switch.py:598]: Shouldn't change this. The test relies on this being 0.1.

File: custom_components/adaptive_lighting/switch.py [hunk 2]
```diff
@@ -1191,7 +1192,15 @@ async def _adapt_lights(
                     context.id,
                 )
                 continue
-            await self._adapt_light(light, transition, force=force, context=context)
+            tasks.append(
+                asyncio.create_task(
+                    self._adapt_light(light, transition, force=force, context=context)
+                )
+            )
+        if len(tasks):
```
Comment (basnijholt) [inline custom_components/adaptive_lighting/switch.py:1200]: just `if tasks:` is the idiomatic way to do this.

File: custom_components/adaptive_lighting/switch.py [hunk 3]
```diff
@@ -1191,7 +1192,15 @@ async def _adapt_lights(
                     context.id,
                 )
                 continue
-            await self._adapt_light(light, transition, force=force, context=context)
+            tasks.append(
+                asyncio.create_task(
```
[the remark this answers is not in the document]
Comment (th3w1zard1) [reply custom_components/adaptive_lighting/switch.py:1196]: Hey maybe you're right, but what's odd is this came from the python3 documentation itself.

[the remark this answers is not in the document]
Comment (th3w1zard1) [reply custom_components/adaptive_lighting/switch.py]: Fixed - still fails the `autoreset_manual_control` test

[the remark this answers is not in the document]
Comment (th3w1zard1) [reply custom_components/adaptive_lighting/switch.py:1279]: Looking at stackoverflow most examples use `asyncio.create_task()` before `asyncio.gather().

**Comment by th3w1zard1** [comment]
I've verified on multiple different stackoverflow threads that this syntax is correct... I'm very confused about how it doesn't pass the test. I even tried increasing the time in `test_manual_control` (correctly this time not https://github.com/basnijholt/adaptive-lighting/pull/529/commits/87ea5243b97bfa064c282382b29e09ed61917892 where I modified the wrong value)

With this issue and the issue from 1.10.0, I believe there's an unforeseen issue with `_AsyncSingleShotTimer`. We still haven't completely resolved the issue from 1.10.0: it's supposed to call the `reset()` function when the timer finishes yet the transition timer's `reset()` function is never called.

**Comment by th3w1zard1** [comment]
Fixed. `hass.async_create_task` was the solution. It now passes the pytest.

File: custom_components/adaptive_lighting/switch.py [hunk 4]
```diff
@@ -1316,7 +1316,9 @@ async def _update_manual_control_and_maybe_adapt(
                 else:
                     _fire_manual_control_event(self, light, context)
             else:
-                await self._adapt_light(light, transition, force=force, context=context)
+                self.hass.async_create_task(
```
Comment (basnijholt) [inline custom_components/adaptive_lighting/switch.py:1319]: This whole PR is now these 3 lines change. 

Why does this make it adapt the lights simultaneously?

Comment (serin113) [reply custom_components/adaptive_lighting/switch.py:1322]: The 3-line changes the behavior from adapting each light in the loop* and `await`ing one by one, to *starting adapt tasks as parallel threads* one by one.

The `await` adds enough delay to be noticeable, and more so if the lights don't respond fast enough:
```
>> start adapting light1 -> await response -> [response + communication delay] -> done with light1 -> start adapting light2 -> etc.
```

Making it parallel minimizes that delay as much as possible by not waiting for each light to finish the `_adapt_light` task:
```
>> start adapting light1 -> [response + communication delay] -> done with light1
>> start adapting light2 -> [response + communication delay] -> done with light2
etc.
```


*the loop being the line `for light in filtered_lights:`


(Came across this issue while researching about that small delay I was experiencing with multiple lights)

Comment (th3w1zard1) [reply custom_components/adaptive_lighting/switch.py:1322]: Well said, to add to this the change is most visible in scenarios where the adapt_delay is nonzero

**Comment by basnijholt** [comment]
Sorry for the delay!

This is great. I just made a small change such that the errors are preserved.

**Review by basnijholt — APPROVED** [review]

**Pull request merged**